lib/api/include/irods/check_auth_credentials.h

```diff
@@ -0,0 +1,134 @@
+#ifndef IRODS_CHECK_AUTH_CREDENTIALS_H
+#define IRODS_CHECK_AUTH_CREDENTIALS_H
+
+/// \file
+
+struct RcComm;
+
+/// The input data type used to invoke #rc_check_auth_credentials.
+///
+/// \since 4.3.1
+typedef struct CheckAuthCredentialsInput // NOLINT(modernize-use-using)
+{
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+    /// The part of a fully-qualified iRODS username preceding the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the zone
+    ///
+    /// \since 4.3.1
+    char username[64];
+
+    /// The part of a fully-qualified iRODS username following the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the username
+    ///
+    /// \since 4.3.1
+    char zone[64];
+
+    /// The expected obfuscated password for the user of interest.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed 250 - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    ///
+    /// \since 4.3.1
+    char password[250];
+
+    // NOLINTEND(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+    // NOLINTNEND(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+} checkAuthCredsInp_t;
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define CheckAuthCredentialsInput_PI "str username[64]; str zone[64]; str password[250];"
+
+/// The output data type used to invoke #rc_check_auth_credentials.
+///
+/// \since 4.3.1
+typedef struct CheckAuthCredentialsOutput // NOLINT(modernize-use-using)
+{
+    /// An integer indicating whether the credentials for a particular user are correct
+    /// or not.
+    ///
+    /// If the credentials are correct, this member will be set to 1. Otherwise, it is
+    /// set to 0.
+    int would_succeed;
```

Comment:
A couple of alternative name suggestions...
 - credentials_correct
 - auth_would_succeed

Reply:
Thinking about how all of these would read in code.
```cpp
checkAuthCredsOut_t* output;

if (1 == output->would_succeed)       { ... }
if (1 == output->credentials_correct) { ... }
if (1 == output->auth_would_succeed)  { ... }
```

I can see any of these working, but the suggestions are starting to feel superior from a readability standpoint.

`credentials_correct` stand out the most.

Will think about this one some more. Are there other things we need to consider (for the future) that may affect the naming?

Reply:
Here's one more alternative.
```cpp
checkAuthCredsOut_t* output;

if (IRODS_CREDS_CORRECT   == output->result) { ... }
if (IRODS_CREDS_INCORRECT == output->result) { ... }
if (IRODS_CREDS_NOT_SET   == output->result) { ... }
```

Thoughts?

Reply:
I don't like shortened 'CREDS' - we can use all the letters, if we choose this option.

One vote for 'credentials_correct'.

Reply:
Discush!

Reply:
Done